nsv13/code/modules/munitions/ammunition/pdc_ammo.dm

```diff
@@ -6,16 +6,54 @@
 	ammo_type = /obj/item/ammo_casing/pdc
 	caliber = "mm30.12"
 	max_ammo = 300
-
-
-/obj/item/ammo_box/magazine/pdc/update_icon()
-	if(ammo_count() > 10)
-		icon_state = initial(icon_state)
-	else
-		icon_state = "[initial(icon_state)]_empty"
+	var/fancy_start = 25	//the pixel position of the first 'bullet' shown in the magazine
+	var/fancy_end   = 8		//the pixel position of the last 'bullet'
+	var/last_pixel	//the last pixel position to be used
 
 /obj/item/ammo_casing/pdc
 	name = "30.12x82mm bullet casing"
 	desc = "A 30.12x82mm bullet casing."
 	projectile_type = /obj/item/projectile/bullet/pdc_round
 	caliber = "mm30.12"
+
+/obj/item/ammo_box/magazine/pdc/forceMove()
+	..()
+	update_icon()//update icon when removed from a machine
+
+/obj/item/ammo_box/magazine/pdc/update_icon()
+	if (istype(loc, /obj))//don't update if it's in an object
+		if (istype(loc, /obj/item/storage)||istype(loc, /obj/item/clothing))//unless it's a backpack or something
+			fancy_icon()
+		return FALSE
+	fancy_icon()
+
+/obj/item/ammo_box/magazine/pdc/examine(mob/user)
+	. = ..()
+	. += "<span class ='notice'>It has [ammo_count()] bullets left.</span>"
+
+/obj/item/ammo_box/magazine/pdc/proc/fancy_icon()//new proc because it's probably too slow for all the times update_icon gets called
+	var/count = stored_ammo.len //faster than ammo_count()
+	var/new_pixel = round(((count/max_ammo)*(fancy_start - fancy_end)) + fancy_end + 1)
+
+	if (!count)
+		icon_state = "[initial(icon_state)]_empty"
+		cut_overlays()
+		last_pixel = null
+		return FALSE
+	icon_state = "[initial(icon_state)]"
+
+	if (new_pixel == last_pixel) 
+		return FALSE
+	last_pixel = new_pixel
+	cut_overlays()
+
+	if (count >= max_ammo)
+		return FALSE
+	
+	var/icon/mask = icon("[icon]","[initial(icon_state)]_empty")
+	mask.Crop(new_pixel,1,(32 + new_pixel),32)
+	mask.Shift(EAST,(new_pixel - 1))
+	add_overlay(mask)
+	return TRUE
```

Comment:
I did it this way because I didn't want to add 17 different icon states per box of ammo (or any icon states at all)
And I didn't want to edit non-NSV files.
I could think of a way of doing it with just one added icon state if you don't want me to use Crop() for some reason
And yeah, I'll move everything into update_icon()

Reply:
runtime image manipulation should be avoided where possible, it's relatively expensive for what you're achieving

Reply:
that's already been changed